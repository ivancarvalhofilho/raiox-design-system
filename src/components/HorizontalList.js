import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const ListContainer = styled.div`
  display: flex;
  position: relative;
  width: 100%;
  padding-top: 16px;
  padding-bottom: 16px;
  overflow-x: hidden;
  font-family: inherit;
`
const List = styled.ul`
  list-style: none;
  padding: 0;
  margin: 0;
  display: grid;
  gap: 0;
  max-width: 100%;
  grid-auto-flow: column;
  padding-bottom: ${({ scroll }) => scroll ? '16px' : 0};
  overflow-x: auto;

  &::-webkit-scrollbar-track{
	-webkit-box-shadow: inset 0 0 6px rgba(0,0,0,0.3);
    background-color: #edeaea;
  }

  &::-webkit-scrollbar-track-piece{
	-webkit-box-shadow: inset 0 0 1px rgba(0, 0, 0, 0.13);
    background-color: #F2F2F2;
  }
  &::-webkit-scrollbar{
    height: 14px;
    background-color: #F2F2F2;
  }
  &::-webkit-scrollbar-thumb{
    border-radius: 4px;
    -webkit-box-shadow: 
      inset 0px 3.5px 0px #F2F2F2,
      inset 0px -3.5px 0px #F2F2F2;
    background-color: rgba(64, 185, 187, 0.4);
  }

  &.active {
    background: rgba(255,255,255,0.3);
    cursor: grabbing;
    cursor: -webkit-grabbing;
    transform: scale(1.02);
  }
`
const ListItem = styled.li`
  display: flex;
  flex-direction: column;
  position: relative;
  width: calc(160px + 96px);
  padding: 2px 48px 12px;
  border-left: 1px solid #E0E0E0;
  box-sizing: border-box;

  &:first-child {
    border-left: none;
    padding-left: 0;
    width: calc(160px + (96px*.5));
  }
`
const Title = styled.p`
  margin: 0;
  color: #595A5C;
  font-size: 13px;
  font-weight: 400;
  line-height: 11px; /* 84.615% */
  letter-spacing: 0.6px;
  margin-bottom: 4px;
`
const Value = styled.p`
  margin: 0;
  color: #303133;
  font-size: 14px;
  font-weight: 500;
  line-height: 22px; /* 157.143% */
  max-width: 160px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  margin-bottom: 8px;
`

/**
 * @typedef {[string,string]} LabelsKey - Array contento na primeira posição a label(título) a ser exibida e segunda posição a chave do objeto com valor
 * @template T
 * @typedef {{
 *  [Property in keyof T]: string | LabelsKey[];
 * } & {labelsKey: LabelsKey[];}} List - Objeto com valores a serem exibidos e seu array de labelsKey
 * @param {{list: List[] | any[]}} props 
 * @example <HorizontalList list={[{
 *    nome: 'Horti Fruti',
      cnpj: "12323132132112",
      codigoFiliacao: "324543552",
      labelsKey: [['Estabelecimento:','nome'],
        ['CNPJ:', 'cnpj'],
        ['Código filiação', 'codigoFiliacao']],
 * }]} />
 * @returns {React.JSX.Element}
 */
const HorizontalList = (props) => {
  const slider = React.useRef(null)
  let isDown = false
  let startX
  let scrollLeft

  const onMouseDown = (e) => {
    isDown = true
    slider.current && slider.current.classList.add('active')
    startX = slider.current && (e.pageX - slider.current.offsetLeft)
    scrollLeft = slider.current && slider.current.scrollLeft
  }

  const onMouseUp = () => {
    isDown = false
    slider.current && slider.current.classList.remove('active')
  }

  const onMouseMove = (e) => {
    if (!isDown) return
    e.preventDefault()
    const x = slider.current && (e.pageX - slider.current.offsetLeft)
    const walk = x && (x - startX) * 1.6
    if (slider.current && walk)
      slider.current.scrollLeft = scrollLeft - walk
  }

  const onMouseLeave = () => {
    isDown = false
    slider.current && slider.current.classList.remove('active')
  }

  return (
    <ListContainer>
      <List
        ref={slider}
        onMouseDown={onMouseDown}
        onMouseUp={onMouseUp}
        onMouseMove={onMouseMove}
        onMouseLeave={onMouseLeave}
        scroll={props.list.length >= 4}
      >
        {
          props.list.map((item, index) => (
            <ListItem key={`estab${index}`}>
              {item.labelsKey.map((key, i) => (
                <div key={`labels${i}`}>
                  <Title>{key[0]}</Title>
                  <Value>{item[key[1]]}</Value>
                </div>
              ))}
            </ListItem>
          ))
        }

      </List>
    </ListContainer>
  )
}

HorizontalList.propTypes = {
  list: PropTypes.array,
}

export { HorizontalList }