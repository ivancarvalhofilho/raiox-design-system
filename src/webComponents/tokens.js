import {Tokens} from "../tokens";

window.customColors = Tokens.coreColors
window.icons = Tokens.icons
