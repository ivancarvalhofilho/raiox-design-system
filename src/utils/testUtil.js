import { ThemeProvider } from 'styled-components'
import { Tokens } from '../tokens'
import React from 'react'
import { mount } from 'enzyme'
import 'jest-styled-components'

export function mountWithTheme(child) {
  return mount(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider Tokens={Tokens}>{children}</ThemeProvider>
    ),
  })
}

export function camelize(str) {
  return str.replace(/(?:^\w|[A-Z]|\b\w)/g, function(word, index) {
    return index === 0 ? word.toLowerCase() : word.toUpperCase();
  }).replace(/\s+/g, '').replace(/\W/gm,'');
}
