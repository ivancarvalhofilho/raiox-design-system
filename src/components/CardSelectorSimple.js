import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import {Tokens} from '../tokens'
import {fontStyleMaker} from '../utils'
import {Icon} from "./Icon";


const CardSelectorSimpleStyle = styled.div`
  transition: 0.15s;
  background: ${Tokens.colors.neutral.light.base};
  box-shadow: 0 0 0 1px ${Tokens.colors.neutral.dark["03"]};
  border-radius: ${Tokens.border.radius.sm};
  user-select: none;
  cursor: pointer;

  pointer-events: ${props => props.disabled && 'none'};
  opacity: ${props => props.disabled && Tokens.opacity.level.medium};
  padding: 16px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 8px;
  margin-top: 8px;
`

const Title = styled.span`
  ${fontStyleMaker({
    fontFamily: "body",
    fontWeight: "regular",
    fontSize:"xs"
  })};
  color: ${Tokens.colors.brand.secondary.dark};
`
const CardSelectorSimple = props => (
    <CardSelectorSimpleStyle
        disabled={props.disabled}
        onClick={() => props.onClick()}
        style={props.style}
        className={props.className}
    >
        <Title>{props.title}</Title>
        {props.arrowIcon &&
        <Icon path={Tokens.icons["single-arrow-right"]} size='md'/>
        }
        {(!props.arrowIcon && props.rightIcon) && (props.rightIcon)}
    </CardSelectorSimpleStyle>
)

CardSelectorSimple.defaultProps = {
  arrowIcon: true,
}

CardSelectorSimple.propTypes = {
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  title: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.string,
  ]),
  style: PropTypes.any,
  className: PropTypes.any,
  arrowIcon: PropTypes.bool,
  rightIcon: PropTypes.element,
}

export {CardSelectorSimple}