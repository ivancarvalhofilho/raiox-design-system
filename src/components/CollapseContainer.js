import PropTypes from 'prop-types'
import React, {useRef, useState} from 'react'
import {Icon} from './Icon'
import styled from 'styled-components'
import {UnmountClosed} from 'react-collapse'
import {
    Container as StyledContainer,
    Divider as DividerStyled,
} from './StyledComponents'
import {Tokens} from "../tokens";
import {fontStyleMaker, handleOutsideDivClick} from "../utils";

const Container = styled(StyledContainer)`
  box-shadow: ${Tokens.shadow.level0};
  padding: 0;
  background: white;
  margin-top: ${Tokens.spacing.inline.xxxs};
  margin-bottom: ${Tokens.spacing.inline.xxxs};
`
const Header = styled.div`
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: ${Tokens.spacing.inset.xs};
`
const Button = styled.div`
  display: flex;
  padding-left: ${Tokens.spacing.inline.nano};
`

const Divider = styled(DividerStyled)`
  width: 90%;
  margin: auto;
  position: relative;
`
const Body = styled.div`
  ${fontStyleMaker({
    fontFamily: "body",
    fontWeight: "regular",
    fontSize: "xs"
  })};
  line-height: 21px;
  color: ${Tokens.colors.neutral.dark.base};
  padding: ${Tokens.spacing.inset.xs};
`
const Title = styled.span`
  ${fontStyleMaker({
    fontFamily: "body",
    fontWeight: "regular",
    fontSize: "xs",
  })};
  line-height: 21px;
  color: ${Tokens.colors.brand.secondary.dark};
`
const CollapseContainer = props => {
    const [opened, setOpened] = useState(false)

    const openRef = useRef()
    openRef.current = opened

    const wrapperRef = handleOutsideDivClick(
        () => {
            if (openRef.current){
                setTimeout(() => {
                    setOpened(false)
                }, 50)
            }
        },
    )

    return (
        <Container ref={wrapperRef}>
            <Header id="header" onClick={() => setOpened(!opened)}>
                <Title>{props.title}</Title>
                <Button>
                    <Icon
                        path={Tokens.icons['chevron-down']}
                        size="16px"
                        rotate={opened ? 180 : 0}
                        appearance={opened ? 'primary' : 'default'}
                    />
                </Button>
            </Header>
            <UnmountClosed isOpened={opened} style={{padding: 0}}>
                <Divider horizontal/>
                <Body id="children">{props.children}</Body>
            </UnmountClosed>
        </Container>
    )
}

export {CollapseContainer}
CollapseContainer.propTypes = {
    children: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
    title: PropTypes.string,
}
