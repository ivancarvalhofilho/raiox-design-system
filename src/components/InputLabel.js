import PropTypes from 'prop-types'
import React, {useRef, useState} from 'react'
import styled from 'styled-components'
import {Tokens} from '../tokens'
import {fontStyleMaker} from '../utils/FontUtil'
import {Icon} from "./Icon";

const SuspendedLabel = styled.div`
  position: absolute;
  z-index: 1;
  margin: 12px;
  padding: 4px;
  color: ${Tokens.colors.neutral.dark['01']};
  ${fontStyleMaker({
	fontFamily: 'body',
	fontWeight: 'regular',
	fontSize: 'xs',
})};
  background: inherit;
  transition: 0.25s;
  top: 0;
  pointer-events: none;
  user-select: none;
  ${(props) => props.hasText && makeLabelSuspended()}
`

function makeLabelSuspended() {
	return {
		top: '-22px',
		...fontStyleMaker({
			fontFamily: 'body',
			fontWeight: 'regular',
			fontSize: 'xxs',
		}),
	}
}

const InputLabelStyled = styled.input`
  padding: 16px;
  box-shadow: inset 0 0 0 1px ${Tokens.colors.neutral.dark['03']};
  border: none;
  box-sizing: border-box;
  border-radius: 4px;
  background: ${Tokens.colors.neutral.light.base};
  transition: 0.25s;
  color: ${Tokens.colors.neutral.dark.base};
  ${fontStyleMaker({
	fontFamily: 'body',
	fontWeight: 'regular',
	fontSize: 'xs',
})};
  width: inherit;
  &:focus {
    outline: none;
    box-shadow: inset 0 0 0 2px ${Tokens.colors.brand.primary.darkest};
  }

  &:focus + div {
    ${makeLabelSuspended()};
  }
  ${props => props.isPasswordType && `
    padding-right: 100px;
  `}
  ${(props) =>
	props.hasError &&
	`
		box-shadow: inset 0 0 0 2px ${Tokens.colors.feedback.danger.dark} !important;
	`}

  &:-webkit-autofill {
    -webkit-background-clip: text;
  }
`
const InputLabelContainer = styled.div`
  background: ${Tokens.colors.neutral.light.base};
  ${(props) =>
	props.disabled &&
	`
    opacity:${Tokens.opacity.level.medium};
    pointer-events: none;
  `};
  position: relative;
  width: 100%;
`
const SpanError = styled.span`
  position: relative;
  display: flex;
  color: ${Tokens.colors.feedback.danger.dark};
  ${fontStyleMaker({
	fontWeight: 'regular',
	fontFamily: 'body',
	fontSize: 'xs',
})};
  margin-top: ${Tokens.spacing.inline.nano};
	text-align: start;
`
const PasswordEyeContainer = styled.div`
  display: flex;
  top: 0;
  right: 0;
  position: absolute;
  align-items: center;
  justify-content: center;
  & > span {
    ${fontStyleMaker({
	fontFamily: "body",
	fontSize: "xxs",
	fontWeight: "regular"
})};
    margin-left: ${Tokens.spacing.stack.nano};
    color: ${Tokens.colors.neutral.dark["01"]};
  }
  height: 48px;
  margin-right: ${Tokens.spacing.stack.xxxs};

  cursor: pointer;
  user-select: none;
`
const InputLabel = (props) => {
	const [visiblePassword, setVisiblePassword] = useState(false)

	let inputType = props.inputType || 'text'

	if (inputType === 'password') {
		inputType = visiblePassword ? 'text' : 'password'
	}

	const isPasswordType = props.inputType === 'password';

	function camelize(str) {
		return str.replace(/(?:^\w|[A-Z]|\b\w)/g, function(word, index) {
			return index === 0 ? word.toLowerCase() : word.toUpperCase();
		}).replace(/\s+/g, '');
	}

	function getTestID() {
		return props.testID || (
			camelize(props.label).replace(/\W/gm,'')
		);
	}

	return (
		<InputLabelContainer
			disabled={props.disabled}
			style={props.style}
			className={props.className}
		>
			<InputLabelStyled
				type={inputType}
				value={props.text}
				hasError={props.error}
				maxLength={props.maxLength}
				onChange={(text) => {
					props.setText(text.target.value)
				}}
				autoFocus={props.autofocus}
				name={props.label}
				id={props.label}
				onKeyDown={(event) => {
					if (!!props.submitOnEnter && event.key === 'Enter') {
						props.submitOnEnter()
					}
				}}
				onBlur={props.onBlur}
				isPasswordType={isPasswordType}
				testID={getTestID()}
			/>
			<SuspendedLabel hasText={!!props.text}>{props.label}</SuspendedLabel>
			{props.error && <SpanError>{props.errorLabel}</SpanError>}
			{isPasswordType && <PasswordEyeContainer onClick={() => setVisiblePassword(!visiblePassword)}>
				<Icon path={visiblePassword ? Tokens.icons["eye-closed"] : Tokens.icons.eye}
							appearance={props.error ? 'danger' : 'primary'} size={20}/>
				<span>{visiblePassword ? 'Esconder' : 'Mostrar'}</span>
			</PasswordEyeContainer>}
		</InputLabelContainer>
	)
}

export {InputLabel}
InputLabel.propTypes = {
	state: PropTypes.string,
	text: PropTypes.string,
	setText: PropTypes.func,
	onBlur: PropTypes.func,
	maxLength: PropTypes.number,
	submitOnEnter: PropTypes.func,
	disabled: PropTypes.bool,
	error: PropTypes.bool,
	errorLabel: PropTypes.string,
	inputType: PropTypes.string,
	autofocus: PropTypes.bool,
	testID: PropTypes.string,
}
