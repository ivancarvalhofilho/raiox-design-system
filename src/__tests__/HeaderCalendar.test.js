import { mountWithTheme } from '../utils/testUtil'
import toJson from 'enzyme-to-json'
import React from 'react'
import { HeaderCalendar } from '../components'
import { Tokens } from '../tokens'

it('should render', () => {
  const spy = jest.spyOn(global.console, 'error')
  const props = {
    month: 0,
    year: 2020,
    disabledLeft: true,
    onBackMonthClick: () => {},
    onNextMonthClick: () => {},
    onMonthClick: () => {},
    values: [
      {
        label: 'Recebido',
        value: '2.502,16',
        color: Tokens.colors.feedback.success.dark,
      },
      {
        label: 'Previsto',
        value: '8.660,07',
        color: Tokens.colors.feedback.danger.dark,
      },
    ],
  }

  const app = mountWithTheme(
    <HeaderCalendar {...props}>TestChildren</HeaderCalendar>,
  )
  expect(toJson(app)).toMatchSnapshot()
  expect(spy).not.toHaveBeenCalled()
})
it('should navigate months', () => {
  const spy = jest.spyOn(global.console, 'error')
  const onBackMonthClick = jest.fn()
  const onNextMonthClick = jest.fn()
  const onMonthClick = jest.fn()
  const props = {
    month: 0,
    year: 2020,
    disabledLeft: false,
    onBackMonthClick: onBackMonthClick,
    onNextMonthClick: onNextMonthClick,
    onMonthClick: onMonthClick,
    values: [
      {
        label: 'Recebido',
        value: '2.502,16',
        color: Tokens.colors.feedback.success.dark,
      },
      {
        label: 'Previsto',
        value: '8.660,07',
        color: Tokens.colors.feedback.danger.dark,
      },
    ],
  }

  const app = mountWithTheme(
    <HeaderCalendar {...props}>TestChildren</HeaderCalendar>,
  )
  const nextButton = app.find('#nextMonth')
  const dateButton = app.find('#date')
  const backButton = app.find('#backMonth')
  expect(onNextMonthClick).not.toHaveBeenCalled()
  nextButton.first().simulate('click')
  expect(onNextMonthClick).toHaveBeenCalled()
  expect(onBackMonthClick).not.toHaveBeenCalled()
  backButton.first().simulate('click')
  expect(onBackMonthClick).toHaveBeenCalled()
  expect(onMonthClick).not.toHaveBeenCalled()
  dateButton.first().simulate('click')
  expect(onMonthClick).toHaveBeenCalled()
  expect(spy).not.toHaveBeenCalled()
})

it('should navigate months with disabled navigate', () => {
  const spy = jest.spyOn(global.console, 'error')
  const onBackMonthClick = jest.fn()
  const onNextMonthClick = jest.fn()
  const onMonthClick = jest.fn()
  const props = {
    month: 0,
    year: 2020,
    disabledLeft: true,
    disabledRight: true,
    onBackMonthClick: onBackMonthClick,
    onNextMonthClick: onNextMonthClick,
    onMonthClick: onMonthClick,
    values: [
      {
        label: 'Recebido',
        value: '2.502,16',
        color: Tokens.colors.feedback.success.dark,
      },
      {
        label: 'Previsto',
        value: '8.660,07',
        color: Tokens.colors.feedback.danger.dark,
      },
    ],
  }

  const app = mountWithTheme(
    <HeaderCalendar {...props}>TestChildren</HeaderCalendar>,
  )
  const nextButton = app.find('#nextMonth')
  const dateButton = app.find('#date')
  const backButton = app.find('#backMonth')
  expect(onNextMonthClick).not.toHaveBeenCalled()
  nextButton.first().simulate('click')
  expect(onNextMonthClick).not.toHaveBeenCalled()
  expect(onBackMonthClick).not.toHaveBeenCalled()
  backButton.first().simulate('click')
  expect(onBackMonthClick).not.toHaveBeenCalled()
  expect(onMonthClick).not.toHaveBeenCalled()
  dateButton.first().simulate('click')
  expect(onMonthClick).toHaveBeenCalled()
  expect(spy).not.toHaveBeenCalled()
})
