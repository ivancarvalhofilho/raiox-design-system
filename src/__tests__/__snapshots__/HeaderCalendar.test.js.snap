// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-transform: rotate(-deg);
  -ms-transform: rotate(-deg);
  transform: rotate(-deg);
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  cursor: pointer;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-transform: rotate(-deg);
  -ms-transform: rotate(-deg);
  transform: rotate(-deg);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 30%;
}

.c4 {
  margin-left: 8px;
  font-family: Roboto;
  font-weight: 500;
  font-size: 18px;
  color: #303133;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
}

.c0 {
  border: 1px solid #e0e0e0;
  box-sizing: border-box;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 56px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  width: Calc(70% - 24px);
  margin-left: 24px;
}

.c6 {
  font-family: Roboto;
  font-weight: 500;
  font-size: 16px;
  color: #303133;
  color: #2E7D32;
  white-space: nowrap;
  margin-left: 24px;
}

.c9 {
  font-family: Roboto;
  font-weight: 500;
  font-size: 16px;
  color: #303133;
  color: #D32F2F;
  white-space: nowrap;
  margin-left: 24px;
}

.c8 {
  font-family: Roboto;
  font-weight: 300;
  font-size: 24px;
  color: #303133;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c7 {
  font-family: Roboto;
  font-weight: 400;
  font-size: 16px;
  color: #303133;
  padding: 0 4px 0 8px;
  white-space: nowrap;
}

<HeaderCalendar
  disabledLeft={true}
  month={0}
  onBackMonthClick={[Function]}
  onMonthClick={[Function]}
  onNextMonthClick={[Function]}
  values={
    Array [
      Object {
        "color": "#2E7D32",
        "label": "Recebido",
        "value": "2.502,16",
      },
      Object {
        "color": "#D32F2F",
        "label": "Previsto",
        "value": "8.660,07",
      },
    ]
  }
  year={2020}
>
  <styled.div>
    <div
      className="c0"
    >
      <styled.div>
        <div
          className="c1"
        >
          <Icon
            appearance="primary"
            disabled={true}
            id="backMonth"
            onClick={[Function]}
            path="<svg width=\\"9\\" height=\\"14\\" viewBox=\\"0 0 9 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M9 12.3594L3.72785 7L9 1.64062L7.38608 0L0.5 7L7.38608 14L9 12.3594Z\\"/></svg>"
            size="16px"
            style={
              Object {
                "cursor": false,
                "marginRight": "8px",
                "opacity": 0.4,
              }
            }
          >
            <Styled(SVGInline)
              appearance="primary"
              disabled={true}
              fill="#008488"
              id="backMonth"
              onClick={[Function]}
              path="<svg width=\\"9\\" height=\\"14\\" viewBox=\\"0 0 9 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M9 12.3594L3.72785 7L9 1.64062L7.38608 0L0.5 7L7.38608 14L9 12.3594Z\\"/></svg>"
              size="16px"
              style={
                Object {
                  "cursor": false,
                  "marginRight": "8px",
                  "opacity": 0.4,
                }
              }
              svg="<svg width=\\"9\\" height=\\"14\\" viewBox=\\"0 0 9 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M9 12.3594L3.72785 7L9 1.64062L7.38608 0L0.5 7L7.38608 14L9 12.3594Z\\"/></svg>"
              width="16px"
            >
              <SVGInline
                appearance="primary"
                className="c2"
                classSuffix="-svg"
                cleanup={Array []}
                cleanupExceptions={Array []}
                component="span"
                disabled={true}
                fill="#008488"
                id="backMonth"
                onClick={[Function]}
                path="<svg width=\\"9\\" height=\\"14\\" viewBox=\\"0 0 9 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M9 12.3594L3.72785 7L9 1.64062L7.38608 0L0.5 7L7.38608 14L9 12.3594Z\\"/></svg>"
                size="16px"
                style={
                  Object {
                    "cursor": false,
                    "marginRight": "8px",
                    "opacity": 0.4,
                  }
                }
                svg="<svg width=\\"9\\" height=\\"14\\" viewBox=\\"0 0 9 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M9 12.3594L3.72785 7L9 1.64062L7.38608 0L0.5 7L7.38608 14L9 12.3594Z\\"/></svg>"
                width="16px"
              >
                <span
                  appearance="primary"
                  className="SVGInline c2"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<svg class=\\"SVGInline-svg sc-bdfBwQ-svg c2-svg\\" fill=\\"#008488\\" style=\\"width: 16px;height: 16px;\\" width=\\"9\\" height=\\"14\\" viewBox=\\"0 0 9 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M9 12.3594L3.72785 7L9 1.64062L7.38608 0L0.5 7L7.38608 14L9 12.3594Z\\"/></svg>",
                    }
                  }
                  disabled={true}
                  id="backMonth"
                  onClick={[Function]}
                  path="<svg width=\\"9\\" height=\\"14\\" viewBox=\\"0 0 9 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M9 12.3594L3.72785 7L9 1.64062L7.38608 0L0.5 7L7.38608 14L9 12.3594Z\\"/></svg>"
                  size="16px"
                  style={
                    Object {
                      "cursor": false,
                      "marginRight": "8px",
                      "opacity": 0.4,
                    }
                  }
                />
              </SVGInline>
            </Styled(SVGInline)>
          </Icon>
          <Icon
            appearance="primary"
            id="nextMonth"
            onClick={[Function]}
            path="<svg width=\\"9\\" height=\\"14\\" viewBox=\\"0 0 9 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M1.08049e-06 1.64063L5.27215 7L1.43428e-07 12.3594L1.61392 14L8.5 7L1.61393 3.51674e-07L1.08049e-06 1.64063Z\\"/></svg>"
            size="16px"
            style={
              Object {
                "cursor": "pointer",
                "opacity": 1,
              }
            }
          >
            <Styled(SVGInline)
              appearance="primary"
              fill="#008488"
              id="nextMonth"
              onClick={[Function]}
              path="<svg width=\\"9\\" height=\\"14\\" viewBox=\\"0 0 9 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M1.08049e-06 1.64063L5.27215 7L1.43428e-07 12.3594L1.61392 14L8.5 7L1.61393 3.51674e-07L1.08049e-06 1.64063Z\\"/></svg>"
              size="16px"
              style={
                Object {
                  "cursor": "pointer",
                  "opacity": 1,
                }
              }
              svg="<svg width=\\"9\\" height=\\"14\\" viewBox=\\"0 0 9 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M1.08049e-06 1.64063L5.27215 7L1.43428e-07 12.3594L1.61392 14L8.5 7L1.61393 3.51674e-07L1.08049e-06 1.64063Z\\"/></svg>"
              width="16px"
            >
              <SVGInline
                appearance="primary"
                className="c3"
                classSuffix="-svg"
                cleanup={Array []}
                cleanupExceptions={Array []}
                component="span"
                fill="#008488"
                id="nextMonth"
                onClick={[Function]}
                path="<svg width=\\"9\\" height=\\"14\\" viewBox=\\"0 0 9 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M1.08049e-06 1.64063L5.27215 7L1.43428e-07 12.3594L1.61392 14L8.5 7L1.61393 3.51674e-07L1.08049e-06 1.64063Z\\"/></svg>"
                size="16px"
                style={
                  Object {
                    "cursor": "pointer",
                    "opacity": 1,
                  }
                }
                svg="<svg width=\\"9\\" height=\\"14\\" viewBox=\\"0 0 9 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M1.08049e-06 1.64063L5.27215 7L1.43428e-07 12.3594L1.61392 14L8.5 7L1.61393 3.51674e-07L1.08049e-06 1.64063Z\\"/></svg>"
                width="16px"
              >
                <span
                  appearance="primary"
                  className="SVGInline c3"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<svg class=\\"SVGInline-svg sc-bdfBwQ-svg c3-svg\\" fill=\\"#008488\\" style=\\"width: 16px;height: 16px;\\" width=\\"9\\" height=\\"14\\" viewBox=\\"0 0 9 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M1.08049e-06 1.64063L5.27215 7L1.43428e-07 12.3594L1.61392 14L8.5 7L1.61393 3.51674e-07L1.08049e-06 1.64063Z\\"/></svg>",
                    }
                  }
                  id="nextMonth"
                  onClick={[Function]}
                  path="<svg width=\\"9\\" height=\\"14\\" viewBox=\\"0 0 9 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M1.08049e-06 1.64063L5.27215 7L1.43428e-07 12.3594L1.61392 14L8.5 7L1.61393 3.51674e-07L1.08049e-06 1.64063Z\\"/></svg>"
                  size="16px"
                  style={
                    Object {
                      "cursor": "pointer",
                      "opacity": 1,
                    }
                  }
                />
              </SVGInline>
            </Styled(SVGInline)>
          </Icon>
          <styled.div
            id="date"
            onClick={[Function]}
          >
            <div
              className="c4"
              id="date"
              onClick={[Function]}
            >
              Janeiro de 2020
            </div>
          </styled.div>
        </div>
      </styled.div>
      <styled.div>
        <div
          className="c5"
        >
          <styled.div
            color="#2E7D32"
          >
            <div
              className="c6"
              color="#2E7D32"
            >
              Recebido
            </div>
          </styled.div>
          <styled.div>
            <div
              className="c7"
            >
              R$
            </div>
          </styled.div>
          <styled.div
            data-tooltip="R$ 2.502,16"
          >
            <div
              className="c8"
              data-tooltip="R$ 2.502,16"
            >
              2.502,16
            </div>
          </styled.div>
          <styled.div
            color="#D32F2F"
          >
            <div
              className="c9"
              color="#D32F2F"
            >
              Previsto
            </div>
          </styled.div>
          <styled.div>
            <div
              className="c7"
            >
              R$
            </div>
          </styled.div>
          <styled.div
            data-tooltip="R$ 8.660,07"
          >
            <div
              className="c8"
              data-tooltip="R$ 8.660,07"
            >
              8.660,07
            </div>
          </styled.div>
        </div>
      </styled.div>
    </div>
  </styled.div>
</HeaderCalendar>
`;
